FE/src/components/SingleRecipe.jsx

```diff
@@ -1,87 +1,573 @@
+                <form onSubmit={(event)=>handleCreateComment(event, reviewId)}>
+                    <label>
+                        Content: <input type='text' value={commentContent} onChange={(e) => setCommentContent(e.target.value)}></input>
+                    </label>
+                    <button>Submit</button>
+                </form>
+                </>
+            )
+        }
+    }
+
+    async function handleCreateComment(event, reviewId){
+        event.preventDefault();
+        async function createCommentFetch(reviewId){
+            try {
+                const response = await fetch(`http://localhost:3000/api/comments/${reviewId}`,
+                { 
+                    method: "POST",
+                    headers: { 
+                        "Content-Type": "application/json",
+                        "Authorization" :`BEARER ${token}`
+                    },
+                    body: JSON.stringify({
+                        content: commentContent
+                    })
+                });
+                const json = await response.json();
+                return json;
+            } catch (error) {
+                setCommentErrMess(reviewId);
+                throw (error);
+            }
+        }
+
+        const potentialComment = await createCommentFetch(reviewId);
+        if (potentialComment.id){
+            setCommentErrMess(null);
+            setRefreshCounter((prev) => prev + 1);
+            setLeavingAComment(null);
+        } else {
+            setCommentErrMess(reviewId);
+        }
+}
+
+
+
+function editingAReviewForm(reviewId){
+    if (editingAReview === reviewId){
+        return (
+            <>
+            <form onSubmit={(event) => handleEditReview(event, reviewId)}>
+                <label>
+                    Title: <input type='text' value={editReviewTitle} onChange={(e) => setEditReviewTitle(e.target.value)}></input>
+                </label>
+                <label>
+                    Rating: <select id="stars" value={editReviewRating} onChange={(e) => setEditReviewRating(e.target.value)} name="stars">
+                            <option value={1}>1</option>
+                            <option value={2}>2</option>
+                            <option value={3}>3</option>
+                            <option value={4}>4</option>
+                            <option value={5}>5</option>
+                            </select>
+                </label>
+                <label>
+                    Content: <input type='text' value={editReviewContent} onChange={(e) => setEditReviewContent(e.target.value)}></input>
+                </label>
+                <button>Submit</button>
+            </form>
+            </>
+        )
+    }
+}
+
+async function handleEditReview(event, reviewId){
+    event.preventDefault();
+    async function editReviewFetch(reviewId){
+        try {
+            const response = await fetch(`http://localhost:3000/api/reviews/${reviewId}`,
```

Comment:
Check with Lisa to see where she was handling the api url, it'd be better to use a variable, so we don't have to go back and change the api url in a bunch of places when we go live.